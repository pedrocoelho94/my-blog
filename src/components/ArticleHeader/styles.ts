import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
