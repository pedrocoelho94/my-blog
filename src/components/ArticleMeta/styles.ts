import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
