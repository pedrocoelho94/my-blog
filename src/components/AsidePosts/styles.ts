import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
