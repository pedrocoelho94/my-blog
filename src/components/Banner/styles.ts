import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  position: relative;
`

export const Content = styled.div`
  filter: brightness(0.95);
  transition: filter 0.2s ease-in-out;

  &:hover {
    filter: brightness(1.2);
  }
`

type imgProps = {
  src: string
}

export const Image = styled.div<imgProps>`
  ${({ src }) => css`
    width: 100%;
    height: 30rem;
    background-image: url(${src});
    background-position: center center;
    background-size: cover;

    ${media.greaterThan('small')`
      height: 40rem;
    `}

    ${media.greaterThan('medium')`
      height: 45rem;
    `}
  `}
`
export const Caption = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.medium};
    background-color: rgba(0, 0, 0, 0.4);
    position: absolute;
    bottom: 0;
    left: 0;
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    margin-top: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      font-size: 2rem;
      margin-bottom: 0;
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-weight: inherit;
    margin-bottom: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      display: none
    `}
  `}
`
