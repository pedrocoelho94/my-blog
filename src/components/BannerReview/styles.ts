import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Poster = styled.div`
  position: relative;
  width: 100%;
  max-width: 25rem;
  height: 100%;
  max-height: 35rem;
  transition: all 0.2s ease-in-out;
  overflow: hidden;
  border-radius: 5px;
`

export const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 5px;
  transition: all 0.2s ease-in-out;
  transform: scale(1.05);

  &:hover {
    transform: scale(1.1);
  }
`

type RatingProps = {
  rating: number
}

const RatingModifiers = {
  good: () => `
    background-color: #01d200;
  `,
  ok: () => `
    background-color: #ffb300;
  `,
  bad: () => `
    background-color: #ff7e00;
  `,
  terrible: () => `
    background-color: #df0700;
  `
}

export const Rating = styled.div<RatingProps>`
  ${({ theme, rating }) => css`
    position: absolute;
    bottom: 0;
    right: 0;
    z-index: 5;
    color: ${theme.colors.white};
    font-size: 1.8rem;
    font-weight: bold;
    width: 3.5rem;
    height: 3.5rem;
    display: flex;
    border-radius: 5px;

    span {
      width: 100%;
      text-align: center;
      justify-self: center;
      align-self: center;
    }

    ${rating >= 7.5 && RatingModifiers.good()}
    ${rating >= 5 && rating < 7.5 && RatingModifiers.ok()}
    ${rating >= 2.5 && rating < 5 && RatingModifiers.bad()}
    ${rating < 2.5 && RatingModifiers.terrible()}
  `}
`
