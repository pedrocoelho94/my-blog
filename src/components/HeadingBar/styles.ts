import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    gap: ${theme.spacings.xsmall};
    background-color: ${theme.colors.secondary};
    color: ${theme.colors.white};
    font-weight: bold;
    padding: ${theme.spacings.xsmall};

    svg {
      height: 22px;
    }
  `}
`
