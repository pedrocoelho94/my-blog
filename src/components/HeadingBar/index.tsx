import * as S from './styles'

export type HeadingBarProps = {
  children: React.ReactNode
  icon?: React.ReactNode
}

const HeadingBar = ({
  children = 'Insira um texto aqui',
  icon
}: HeadingBarProps) => (
  <S.Wrapper>
    {icon} {children}
  </S.Wrapper>
)

export default HeadingBar
