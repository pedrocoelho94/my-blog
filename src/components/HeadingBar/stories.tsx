import { Story, Meta } from '@storybook/react'
import HeadingBar, { HeadingBarProps } from '.'
import { ListAlt } from '@styled-icons/material-outlined/ListAlt'

export default {
  title: 'HeadingBar',
  component: HeadingBar,
  args: {
    children: 'oba',
    icon: <ListAlt />
  }
} as Meta

export const Default: Story<HeadingBarProps> = (args) => (
  <HeadingBar {...args} />
)
