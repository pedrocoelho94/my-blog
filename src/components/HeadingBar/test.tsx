import { renderWithTheme } from 'styles/renderWithTheme'
import { ListAlt } from '@styled-icons/material-outlined/ListAlt'

import HeadingBar from '.'

const mock = {
  children: 'oba',
  icon: <ListAlt />
}

describe('<HeadingBar />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <HeadingBar icon={mock.icon}>{mock.children}</HeadingBar>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
