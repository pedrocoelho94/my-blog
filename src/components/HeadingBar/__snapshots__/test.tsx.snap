// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadingBar /> should render the heading 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  background-color: #dc143c;
  color: #FAFAFA;
  font-weight: bold;
  padding: 0.8rem;
}

.c0 svg {
  height: 22px;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    class="c1"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M11 7h6v2h-6zm0 4h6v2h-6zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7zM20.1 3H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM19 19H5V5h14v14z"
    />
  </svg>
   
  oba
</div>
`
